doc/api/hooks_server-side.md

```diff
@@ -398,22 +412,36 @@ Called from: src/node/handler/PadMessageHandler.js
 Things in context:
 
 1. message - the message being handled
-2. client - the client object from socket.io
+2. client - the socket.io Socket object
 
-This hook will be called once a message arrives. If a plugin calls `callback(true)` the message will be allowed to be processed. This is especially useful if you want read only pad visitors to update pad contents for whatever reason.
+This hook allows plugins to grant temporary write access to a pad. It is called
+for each incoming message from a client. If write access is granted, it applies
+to the current message and all future messages from the same socket.io
+connection until the next `CLIENT_READY` or `SWITCH_TO_PAD` message. Read-only
+access is reset **after** each `CLIENT_READY` or `SWITCH_TO_PAD` message, so
+granting write access has no effect for those message types.
 
-**WARNING**: handleMessageSecurity will be called, even if the client is not authorized to send this message. It's up to the plugin to check permissions.
+The handleMessageSecurity function must return a Promise. If the Promise
+resolves to `true`, write access is granted as described above. Returning
+`callback(value)` will return a Promise that is resolved to `value`.
 
-Example:
+**WARNING:** handleMessageSecurity is called for every message, even if the
+client is not authorized to send the message. It is up to the plugin to check
+permissions.
+
+Examples:
 
 ```
-function handleMessageSecurity ( hook, context, callback ) {
-  if ( context.message.boomerang == 'hipster' ) {
-    // If the message boomer is hipster, allow the request
-    callback(true);
-  }else{
-    callback();
-  }
+// Using an async function:
+exports.handleMessageSecurity = async (hookName, {message, client}) => {
```

Comment:
wouldn't this break every plugin that used this? :)

Reply:
I'm just documenting the current behavior. Currently plugin authors can either write their `handleMessageSecurity` function as a regular function with a callback or as an async function. Either way works.